### ISSUE TITLE
[a11y]: Dropdown and FilterableMultiSelect components do not have aria roles set when using Firefox

### ISSUE THREAD

**Issue by patrickobrien1** [carbon-design-system/carbon#20084]
### Package

@carbon/react

### Browser

Firefox

### Operating System

MacOS

### Package version

1.76.0

### React version

16.14.0

### Automated testing tool and ruleset

Accessibility Audit team

### Assistive technology

VoiceOver

### Description

When using the VoiceOver screen reader, the Dropdown and FilterableMultiSelect components will have their roles stripped for the screen reader when using Firefox rather than Google Chrome as a web browser.

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-dropdown--overview

### Steps to reproduce

<img width="720" height="420" alt="Image" src="https://github.com/user-attachments/assets/60e84e45-fe74-4809-a4f4-9b31483e0d57" />

<img width="720" height="432" alt="Image" src="https://github.com/user-attachments/assets/9caa12c2-12e5-46c2-95d6-95cf78f29588" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
